lib/Doctrine/DBAL/Types/DateIntervalType.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+
+/**
+ * Type that maps interval string to a PHP DateInterval Object.
+ */
+class DateIntervalType extends Type
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return Type::DATEINTERVAL;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
+    {
+        return $platform->getVarcharTypeDeclarationSQL($fieldDeclaration);
```

Comment:
Since the data format's length will never change, the generated column should have a fixed length. See DBAL\Types\GuidType for example of a similar situation.